src/systems/user_commands/UserCommands.cc

```diff
@@ -437,36 +447,24 @@ class WheelSlipCommand : public UserCommandBase
 /// \brief Private UserCommands data class.
 class gz::sim::systems::UserCommandsPrivate
 {
-  /// \brief Callback for create service
-  /// \param[in] _req Request containing entity description.
-  /// \param[out] _res True if message successfully received and queued.
-  /// It does not mean that the entity will be successfully spawned.
-  /// \return True if successful.
-  public: bool CreateService(const msgs::EntityFactory &_req,
-      msgs::Boolean &_res);
-
-  /// \brief Callback for multiple create service
-  /// \param[in] _req Request containing one or more entity descriptions.
-  /// \param[out] _res True if message successfully received and queued.
-  /// It does not mean that the entities will be successfully spawned.
-  /// \return True if successful.
-  public: bool CreateServiceMultiple(
-              const msgs::EntityFactory_V &_req, msgs::Boolean &_res);
-
-  /// \brief Callback for remove service
-  /// \param[in] _req Request containing identification of entity to be removed.
-  /// \param[out] _res True if message successfully received and queued.
-  /// It does not mean that the entity will be successfully removed.
-  /// \return True if successful.
-  public: bool RemoveService(const msgs::Entity &_req,
-      msgs::Boolean &_res);
-
-  /// \brief Callback for light service
-  /// \param[in] _req Request containing light update of an entity.
-  /// \param[out] _res True if message successfully received and queued.
-  /// It does not mean that the light will be successfully updated.
-  /// \return True if successful.
-  public: bool LightService(const msgs::Light &_req, msgs::Boolean &_res);
+  /// \brief Temlpate for handling service calls. This copies the message and
+  /// adds it the pending commands queue.
+  /// \tparam CommandT Type for the command that associated with the service.
+  /// \tparam InputT Type form gz::msgs of the input parameter.
+  /// \param[in] _req Input parameter message of the service.
+  /// \param[out] _res Output parameter message of the service.
+  public: template <typename CommantT, typename InputT>
```

Comment:
```suggestion
  public: template <typename CommandT, typename InputT>
```
here and elsewhere in this file